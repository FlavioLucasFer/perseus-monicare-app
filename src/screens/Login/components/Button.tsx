import React from 'react';
import { 
	ActivityIndicator,
	StyleSheet, 
	Text, 
	TouchableOpacity, 
	TouchableOpacityProps, 
} from 'react-native';

interface LoginButtonProps extends TouchableOpacityProps {
	title: string,
	loading?: boolean,
};

const styles = StyleSheet.create({
	container: {
		display: 'flex',
		flexDirection: 'row',
		justifyContent: 'center',
		alignItems: 'center',

		backgroundColor: '#FC9A9A',
		borderWidth: 2,
		borderColor: '#FF1616',
		borderRadius: 50,
		padding: 10,
	},

	disabled: {
		backgroundColor: '#FC9A9A4D',
		borderColor: '#FF16164D',
	},

	title: {
		color: '#000',
		fontSize: 20,
		fontWeight: 'bold',
	},

	titleDisabled: {
		color: '#00000080',
	},
});

const Button: React.FC<LoginButtonProps> = props => {
	const {
		title,
		loading,
		style,
		...rest
	} = props;

	return (
		<TouchableOpacity
			{...rest}
			style={[styles.container, props.disabled && styles.disabled, style]}
		>
			{loading &&
				<ActivityIndicator 
					color='#FF1616'
					style={{ marginRight: 10 }}
				/>
			}
			<Text style={[styles.title, props.disabled && styles.titleDisabled]}>{title}</Text>
		</TouchableOpacity>
	);
};

export type {
	LoginButtonProps,
};
export default Button;
