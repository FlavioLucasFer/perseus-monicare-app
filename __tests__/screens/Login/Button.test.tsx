import React from 'react';
import { shallow } from 'enzyme';

import Button from 'screens/Login/components/Button';

describe('Button component', () => {
	it('Should render properly', () => {
		const wrapper = shallow(
			<Button title='teste' />
		);
		expect(wrapper.debug()).toMatchSnapshot();
	});
});
