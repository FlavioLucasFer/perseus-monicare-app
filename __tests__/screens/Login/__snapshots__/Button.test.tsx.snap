// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Should render properly 1`] = `
"<TouchableOpacity style={{...}}>
  <Text style={{...}}>
    teste
  </Text>
</TouchableOpacity>"
`;
